scripts/renderer.lua

```diff
@@ -188,14 +196,16 @@ end
 
 --- @param entity_data EntityData
 function renderer.clear(entity_data)
-  clear_sprite(entity_data.shape)
-  entity_data.shape = nil
-  for _, objects in pairs(entity_data.connection_objects) do
-    for _, id in pairs(objects) do
-      clear_sprite(id)
+  if entity_data.shape then
+    clear_sprite(entity_data.shape)
+    entity_data.shape = nil
+    for _, objects in pairs(entity_data.connection_objects) do
```

Comment:
This should be outside of the if statement, since an entity can have connections but no shape.